api/v1alpha1/servicedefaults_types.go

```diff
@@ -63,26 +100,17 @@ func init() {
 }
 
 // ToConsul converts the entry into it's Consul equivalent struct.
-func (in *ServiceDefaults) ToConsul() *capi.ServiceConfigEntry {
+func (in *ServiceDefaults) ToConsul() capi.ConfigEntry {
```

Comment:
This needed to change to return the interface so that the `ToConsul` method could be used generically. This doesn't lose any data because when Consul serializes is for its API call it will contain all these fields.